internal/commands/vaultsecrets/run/run.go

```diff
@@ -0,0 +1,176 @@
+        process are forwarded to the top level 'hcp vault-secrets run' command.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Inject secrets as an environment variable:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets run --command="env"
+				`),
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "command",
+					DisplayValue: "COMMAND",
+					Description:  "Defines the invocation of the child process to inject secrets to.",
+					Value:        flagvalue.Simple("", &opts.Command),
+					Required:     true,
+				},
+				{
+					Name:         "app",
+					DisplayValue: "APP",
+					Description:  "The application you want to pull all secrets from.",
+					Value:        appname.Flag(),
+				},
+			},
+		},
+		PersistentPreRun: func(c *cmd.Command, args []string) error {
+			return appname.Require(ctx)
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.App = appname.Get()
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+func runRun(opts *RunOpts) (err error) {
+	envSecrets, err := getAllSecretsForEnv(opts)
+	if err != nil {
+		return fmt.Errorf("failed to run with secrets in app %q: %w", opts.App, err)
+	}
+
+	childProcess, err := setupChildProcess(opts.Command, envSecrets)
+	if err != nil {
+		return fmt.Errorf("failed to run with secrets in app %q: %w", opts.App, err)
+	}
+
+	if err := childProcess.Start(); err != nil {
+		return fmt.Errorf("failed to run with secrets in app %q: %w", opts.App, err)
+
+	}
+
+	sigC := make(chan os.Signal, 1)
+	signal.Notify(sigC, os.Interrupt, syscall.SIGTERM)
+	for {
+		select {
+		case <-sigC:
+			childProcess.Stop()
+			return fmt.Errorf("failed to run with secrets in app %q: %w", opts.App, err)
+		case <-childProcess.ExitCh():
+			return nil
+		}
+	}
+}
+
+func getAllSecretsForEnv(opts *RunOpts) ([]string, error) {
+	params := preview_secret_service.NewOpenAppSecretsParamsWithContext(opts.Ctx)
+	params.OrganizationID = opts.Profile.OrganizationID
+	params.ProjectID = opts.Profile.ProjectID
+	params.AppName = opts.App
+
+	res, err := opts.PreviewClient.OpenAppSecrets(params, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// get existing environment variables detectable by runtime
+	result := os.Environ()
+
+	for _, secret := range res.Payload.Secrets {
+		// we need to append results in case of duplicates we want secrets to override
+		// only supporting static secrets
+		if secret.StaticVersion != nil {
+			result = append(result, strings.ToUpper(secret.Name)+"="+secret.StaticVersion.Value)
+		}
+	}
+
+	return result, nil
+}
+
+func setupChildProcess(command string, envVars []string) (*child.Child, error) {
+	pieces := strings.Split(command, " ")
+	cmd := pieces[0]
+	var args []string
+	if len(pieces) > 1 {
+		args = pieces[1:]
+	}
+
+	input := &child.NewInput{
+		Stdin:       os.Stdin,
+		Stdout:      os.Stdout,
+		Stderr:      os.Stderr,
```

Comment:
@dadgar do we need to use ctx.IO etc. here or since this is a separate child process we are ok with this setup?

Reply:
I am okay making an exception in this case since we are starting a new process. It will be a bit more performant to pass the file descriptors directly.